lib_provisioner/src/main/java/no/nordicsemi/wifi/provisioner/library/internal/response/ResponsePacket.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022, Nordic Semiconductor
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are
+ * permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of
+ * conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list
+ * of conditions and the following disclaimer in the documentation and/or other materials
+ * provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be
+ * used to endorse or promote products derived from this software without specific prior
+ * written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package no.nordicsemi.wifi.provisioner.library.internal.response
+
+import android.bluetooth.BluetoothDevice
+import no.nordicsemi.android.ble.callback.profile.ProfileReadResponse
+import no.nordicsemi.android.ble.data.Data
+import no.nordicsemi.android.wifi.provisioning.Response
+
+internal class ResponsePacket : ProfileReadResponse() {
+    lateinit var value: Response
+
+    override fun onDataReceived(device: BluetoothDevice, data: Data) {
+        super.onDataReceived(device, data)
+
+        try {
+            value = Response.ADAPTER.decode(data.value ?: byteArrayOf())
```

Comment:
I think this try-catch will not propagate exception to UI. This is not good. Please fix.

Reply:
Indeed, there's an issue, but not that there would be no exception.
`suspendForValidResponse()` would throw `InvalidDataException` instead, but the problem is, that this exception is defined in the BLE library. A module using the provisioning library does not need to depend on BLE library, so there can be issues with catching this exception, as its type is not exported by the lib.

Reply:
Please, check the latest changes.

`suspend...()` methods can throw bunch of different exceptions, not only `InvalidDataException`. Also, when the device gets disconnected, Bluetooth is disconnected, etc. Before, all those were re-thrown from the provisioning library, but catching them would require the BLE library. Now, all of them are mapped to `ResponseErrorException(ResponseError.INTERNAL_ERROR)`. I don't know should there be a new exception type defined...